paper/paper.md

```diff
@@ -11,49 +14,57 @@ authors:
+The empirical analysis of algorithms is often accompanied by the execution of algorithms for different inputs and variants of the algorithms (specified via parameters) and the measurement of non-functional properties. Since the individual evaluations are usually independent of each other, the evaluation can be performed in a distributed manner on an HPC system. However, setting up, documenting, and evaluating the results of such a study is often file-based. Usually, this requires extensive manual work to create configuration files for the inputs or to read and aggregate measured results from a report file. In addition, monitoring and restarting individual executions is tedious and time-consuming. 
 
+These challenges are addressed by `PyExperimenter` by means of a single well defined configuration file and a central database for managing massively parallel evaluations, as well as collecting and aggregating their results. Thereby, `PyExperimenter` alleviates the aforementioned overhead and allows experiment executions to be defined and monitored with ease.
 
-# Statement of Need
+![General schema of `PyExperimenter`.](usage.png)
 
-`PyExperimenter` is a Python package supporting easy execution of multiple experiments differing only in their parametrizations. `PyExperimenter` was designed to be used by machine learning researchers and according students, but is not limited to those. The general structure of the project allows using `PyExperimenter` also for other types of experiments, as long as the execution requires the same code parameterized in a different way.  
+A general schema of `PyExperimenter` can be found in Figure 1. `PyExperimenter` is designed based on the assumption that an experiment is uniquely defined by certain inputs, i.e., parameters, and a function computing the results of the experiment based on these parameters. The set of experiments to be executed can be defined through a configuration file listing the domains of each parameter, or manually through code. Those parameters define the experiment grid, based on which `PyExperimenter` setups the table in the database featuring all experiments with their input parameter values and additional information such as the execution status. Once this table has been created, a `PyExperimenter` instance can be run on any machine, including a distributed system. Each instance automatically pulls open experiments from the database, executes the function provided by the user with the corresponding parameters defining the experiment and writes back the results computed by the function. Errors arising during the execution are logged in the database. In case of failed experiments or if desired otherwise, a subset of the experiments can be reset easily and restarted. After all experiments are done, results can be jointly exported as a Pandas DataFrame [@pandas] for further processing, such as generating a LaTeX table averaging over different seeds.
```

Comment:
In case of failed experiments or if desired otherwise, a subset of the experiments can be reset easily and restarted -> In case of failed experiments or if desired otherwise, a subset of the experiments can be easily reset and restarted.

"Thus, in sum, in compound-verb forms having two words, put the adverb between the two verb words"

Reply:
Here, technically both is fine. By postponing the adverb it is given more emphasis.

Reply:
I think it would be best to move 'easily' at the end of the sentence then:

... a subset of the experiments can be reset and restarted easily.

Reply:
agree